c2rust-analyze/src/main.rs

```diff
@@ -440,76 +442,92 @@ fn run(tcx: TyCtxt) {
     // computed during this the process is kept around for use in later passes.
     let mut global_equiv = GlobalEquivSet::new(gacx.num_pointers());
     for &ldid in &all_fn_ldids {
+        // The function might already be marked as failed if one of its callees previously failed.
+        if gacx.fn_failed(ldid.to_def_id()) {
+            continue;
+        }
+
         let ldid_const = WithOptConstParam::unknown(ldid);
         let mir = tcx.mir_built(ldid_const);
         let mir = mir.borrow();
         let lsig = *gacx.fn_sigs.get(&ldid.to_def_id()).unwrap();
 
         let mut acx = gacx.function_context(&mir);
 
-        // Assign PointerIds to local types
-        assert!(acx.local_tys.is_empty());
-        acx.local_tys = IndexVec::with_capacity(mir.local_decls.len());
-        for (local, decl) in mir.local_decls.iter_enumerated() {
-            let lty = match mir.local_kind(local) {
-                LocalKind::Var | LocalKind::Temp => acx.assign_pointer_ids(decl.ty),
-                LocalKind::Arg => {
-                    debug_assert!(local.as_usize() >= 1 && local.as_usize() <= mir.arg_count);
-                    lsig.inputs[local.as_usize() - 1]
-                }
-                LocalKind::ReturnPointer => lsig.output,
-            };
-            let l = acx.local_tys.push(lty);
-            assert_eq!(local, l);
+        let r = panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
please use a more descriptive name

Reply:
I don't think a descriptive name is necessary here, since the value is used only once, immediately after it's defined.